ports/mimxrt10xx/peripherals/mimxrt10xx/pins.h

```diff
@@ -48,22 +48,30 @@ extern const mp_obj_type_t mcu_pin_type;
 typedef struct {
     mp_obj_base_t base;
     GPIO_Type *gpio;
-    uint8_t number;
+    uint8_t port:3; //0 start index
+    uint8_t number:5;
+    uint16_t mux_idx;
     uint32_t mux_reg;
     uint32_t cfg_reg;
     ADC_Type *adc;
     uint8_t adc_channel;
+    uint32_t mux_reset;
+    uint32_t pad_reset;
 } mcu_pin_obj_t;
 
-#define PIN(p_gpio, p_number, p_enum, p_adc, p_adc_channel) \
+#define PIN(p_gpio, p_port, p_number, p_enum, p_adc, p_adc_channel, p_mux_reset, p_pad_reset) \
 { \
     { &mcu_pin_type }, \
     .gpio = p_gpio, \
+    .port = p_port, \
```

Comment:
Could we somehow get around adding an extra, somewhat redundant field by having an array of all gpio ports and then you could use `pin.gpio` to find the port index?

Reply:
We could... it's a little clunky but would still be better than going the STM32 route and having a `pin_port` function that fetches the pointer from the GPIO index every time. I'm thinking my preferred solution however would be to use the mux_idx for everything, rather than bothering with the GPIO banks structurally at all. That way the numbering systems are consistent across Microcontroller, and you only ever need one number as a parameter. I'll take a look at swapping on Monday.